src/Features/CSharp/Portable/CodeLens/DisplayInfoCSharpServices.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics
+{
+    [ExportLanguageService(typeof(IDisplayInfoLanguageServices), LanguageNames.CSharp), Shared]
+    internal sealed class DisplayInfoCSharpServices : IDisplayInfoLanguageServices
+    {
+        private const SymbolDisplayMiscellaneousOptions MiscOptions =
+            SymbolDisplayMiscellaneousOptions.EscapeKeywordIdentifiers |
+            SymbolDisplayMiscellaneousOptions.UseSpecialTypes |
+            SymbolDisplayMiscellaneousOptions.UseAsterisksInMultiDimensionalArrays |
+            SymbolDisplayMiscellaneousOptions.UseErrorTypeSymbolName;
+
+        private const SymbolDisplayMemberOptions MemberOptions =
+            SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeContainingType;
+
+        // Matches default ToDisplayString
+        private static readonly SymbolDisplayFormat DefaultFormat = new SymbolDisplayFormat(
+                globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.OmittedAsContaining,
+                typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces,
+                propertyStyle: SymbolDisplayPropertyStyle.NameOnly,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                memberOptions: MemberOptions,
+                parameterOptions: SymbolDisplayParameterOptions.IncludeParamsRefOut | SymbolDisplayParameterOptions.IncludeType,
+                miscellaneousOptions: MiscOptions);
+
+        // Matches default ToDisplayString except removing global namspace and namespaces
+        private static readonly SymbolDisplayFormat ShortFormat = new SymbolDisplayFormat(
+                globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
+                typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypes,
+                propertyStyle: SymbolDisplayPropertyStyle.NameOnly,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                memberOptions: MemberOptions,
+                parameterOptions: SymbolDisplayParameterOptions.IncludeParamsRefOut | SymbolDisplayParameterOptions.IncludeType,
+                miscellaneousOptions: MiscOptions);
+
+        /// <summary>
+        /// Indicates if the given node is a declaration of some kind of symbol. 
+        /// For example a class for a method declaration.
+        /// </summary>
+        public bool IsDeclaration(SyntaxNode node)
+        {
+            return IsTypeOrNamespaceDeclaration(node) || IsMemberDeclaration(node);
```

Comment:
I wonder if all helpers here are already present in Roslyn. Original codelens code might have had to re-implement these because they might have been internal helpers in Roslyn.